frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx

```diff
@@ -0,0 +1,190 @@
+  const [chartDataLoaded, setChartDataLoaded] = React.useState(false);
+  const [chartName, setChartName] = React.useState('');
+  const [chartHasValues, setChartHasValues] = React.useState(false);
+  const [YAMLData, setYAMLData] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+  const [helmChartEntries, setHelmChartEntries] = React.useState<HelmChartMetaData[]>([]);
+  const helmAction: HelmActionType =
+    chartURL !== 'null' ? HelmActionType.Install : HelmActionType.Upgrade;
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchHelmReleases = async () => {
+      let res;
+      try {
+        res =
+          helmAction === HelmActionType.Install
+            ? await coFetchJSON(`/api/helm/chart?url=${chartURL}`)
+            : await coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+      const data: HelmChart = !_.isEmpty(res.chart) ? res.chart : res;
+      setYAMLData(!_.isEmpty(res.config) ? _.merge(data.values, res.config) : data.values);
+      if (helmAction === HelmActionType.Upgrade) {
+        setChartName(data.metadata.name);
+        setActiveChartVersion(data.metadata.version);
+      }
+      setChartDataLoaded(true);
+      !_.isEmpty(data.values) && setChartHasValues(true);
+    };
+
+    fetchHelmReleases();
+
+    return () => {
+      ignore = true;
+    };
+  }, [chartURL, namespace, releaseName, helmAction]);
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchChartVersions = async () => {
+      let json: HelmRelease;
+
+      try {
+        const response = await coFetch('/api/helm/charts/index.yaml');
+        const yaml = await response.text();
+        json = safeLoad(yaml);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+      setHelmChartEntries(_.get(json, ['entries', chartName]));
+    };
+    helmAction !== HelmActionType.Install && fetchChartVersions();
+    return () => {
+      ignore = true;
+    };
+  }, [chartName, helmAction]);
+
+  const initialValues: HelmInstallUpgradeFormData = {
+    helmReleaseName: releaseName || helmChartName || '',
+    chartValuesYAML: !_.isEmpty(YAMLData) ? safeDump(YAMLData) : undefined,
+    chartVersion: activeChartVersion,
+  };
+
+  const handleSubmit = (values, actions) => {
+    const { helmReleaseName, chartVersion, chartValuesYAML }: HelmInstallUpgradeFormData = values;
+    let valuesObj;
+
+    if (chartValuesYAML) {
+      try {
+        valuesObj = safeLoad(chartValuesYAML);
+      } catch (err) {
+        actions.setStatus({ submitError: `Invalid YAML - ${err}` });
+        return;
+      }
+    }
+    let newChartURL = '';
+    newChartURL = chartVersion && getChartURL(helmChartEntries, chartVersion);
+
+    const payload = {
+      namespace,
+      name: helmReleaseName,
+      ...(helmAction === HelmActionType.Install
```

Comment:
Why don't you also save `chartURL` in formik state? Add it in the initial values and update it from the `ChartVersionDropdown` when user selects a new version. That way you don't have to have any logic here and just use `chartURL` from the formik state here.